frontend/views/containers/chatroom/chat-mentions/RenderMessageWithMarkdown.js

```diff
@@ -21,7 +23,33 @@ const RenderMessageWithMarkdown: any = {
 
     // Turns a dom tree object structure into the equivalent recursive createElement(...) call structure.
     const recursiveCall = (entry: any): any => {
-      if (entry.tagName) {
+      if (entry.tagName === 'ROUTER') {
+        const hasChildren = Array.isArray(entry.children)
+        const route = entry.attributes.route && JSON.parse(entry.attributes.route)
+        const href = route && this.$router.resolve(route).href
+        return createElement(
+          'a',
+          {
+            class: 'link',
+            attrs: {
+              href
+            },
+            on: {
+              click: (e) => {
+                route && this.$router.push(route)
+                e?.preventDefault()
+              },
+              touchhold: (e) => {
+                href && sbp('okTurtles.events/emit', OPEN_TOUCH_LINK_HELPER, href)
+                e?.preventDefault()
+              }
+            }
+          },
+          hasChildren
+            ? entry.children.map(child => recursiveCall(child))
+            : undefined
+        )
+      } else if (entry.tagName) {
```

Comment:
This is to find in-app links which are created by `makeInAppLinkElement` function and create elements in real.

Reply:
@Silver-IT , it looks like this additional block has some duplicate codes with the `else if (entry.tagName)` below.
I would DRY this by including the logic in there like,

```js
if (entry.tagName) {
  const hasChildren = Array.isArray(entry.children)
  const isRouter = entry.tagName === 'ROUTER'

  const elName =  isRouter ? 'a' : entry.tagName.toLowerCase()
  const opts = isRouter
    ? {
          // options you created above
       }
    :  {
          // default options
        }
    
  ....
  
   return createElement(
     elName,
     opts,
     hasChildren
        ? entry.children.map(child => recursiveCall(child))
        : undefined
   )
}
```